politeiawww/cmd/piwww/testrun.go

```diff
@@ -446,15 +470,217 @@ func testUserRoutes(admin testUser, minPasswordLength int) error {
 	return nil
 }
 
-// Execute executes the test run command.
-func (cmd *testRunCmd) Execute(args []string) error {
+// createMDFile returns a File object that was created using a markdown file
+// filled with random text.
+func createMDFile() (*pi.File, error) {
+	var b bytes.Buffer
+	b.WriteString("This is the proposal title\n")
 
-	const (
-		// Comment actions
-		commentActionUpvote   = "upvote"
-		commentActionDownvote = "downvote"
-	)
+	for i := 0; i < 10; i++ {
+		r, err := util.Random(32)
+		if err != nil {
+			return nil, err
+		}
+		b.WriteString(base64.StdEncoding.EncodeToString(r) + "\n")
+	}
 
+	return &pi.File{
+		Name:    "index.md",
+		MIME:    mime.DetectMimeType(b.Bytes()),
+		Digest:  hex.EncodeToString(util.Digest(b.Bytes())),
+		Payload: base64.StdEncoding.EncodeToString(b.Bytes()),
+	}, nil
+}
+
+// newNormalProposal is a wrapper func which creates a proposal by calling
+// newProposal
+func newNormalProposal() (*pi.ProposalNew, error) {
+	return newProposal(false, "")
+}
+
+// newProposal returns a NewProposal object contains randonly generated
+// markdown text and a signature from the logged in user. If given `rfp` bool
+// is true it creates an RFP. If given `linkto` it creates a RFP submission.
+func newProposal(rfp bool, linkto string) (*pi.ProposalNew, error) {
+	md, err := createMDFile()
+	if err != nil {
+		return nil, fmt.Errorf("create MD file: %v", err)
+	}
+	files := []pi.File{*md}
+
+	pm := www.ProposalMetadata{
+		Name: "Some proposal name",
+	}
+	if rfp {
+		pm.LinkBy = time.Now().Add(time.Hour * 24 * 30).Unix()
+	}
+	if linkto != "" {
+		pm.LinkTo = linkto
+	}
+	pmb, err := json.Marshal(pm)
+	if err != nil {
+		return nil, err
+	}
+	metadata := []pi.Metadata{
+		{
+			Digest:  hex.EncodeToString(util.Digest(pmb)),
+			Hint:    pi.HintProposalMetadata,
+			Payload: base64.StdEncoding.EncodeToString(pmb),
+		},
+	}
+
+	sig, err := signedMerkleRoot(files, metadata, cfg.Identity)
+	if err != nil {
+		return nil, fmt.Errorf("sign merkle root: %v", err)
+	}
+
+	return &pi.ProposalNew{
+		Files:     files,
+		Metadata:  metadata,
+		PublicKey: hex.EncodeToString(cfg.Identity.Public.Key[:]),
+		Signature: sig,
+	}, nil
+}
+
+// testPropsWithComments tests the propsal & comments routes
```

Comment:
```suggestion
// testPropsWithComments tests the propsal routes
```

Will do in comment routes PR.